### PR TITLE
add bluetooth get request insert schedule

### PR DESCRIPTION
**Pull request by icecat2012**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -221,6 +221,18 @@ def get(self):
...
+    def get(self):
+        bluetooth_id = self.get_argument("bluetooth_id", default='')
+        #self.write("Hello, {}!".format(bluetooth_id))
+        receive_msg = {}
+        receive_msg = deal_with_bluetooth_id(bluetooth_id)
+        if receive_msg["result"] == "success":
+            self.write("success {}".format(bluetooth_id))
+        else :
+            self.write("fail {}".format(bluetooth_id))
+
```
Comment (Billy4195) [inline server.py]: extra blank line

File: server_api.py [hunk 2]
```diff
@@ -18,12 +19,161 @@
...
+            if pure_data[0] == "bluetooth_available":
+                bluetooth_available = int(pure_data[1])
+        file_pointer.close()
+        
+        #check function available
+        if bluetooth_available==1:
+            return 1
+        else :
+            return 0
+
```
Comment (Billy4195) [inline server_api.py]: extra blank line

File: server_api.py [hunk 3]
```diff
@@ -18,12 +19,161 @@
...
+        
+        #reshap pref_data_type_XX from varchar to int array
+        data_type_array = []
+        if len(pure_result) > 0  and pure_result[0][0] is not None:
+            str_condition = pure_result[0][0].split(' ')
+            for num1 in range(len(str_condition)):
+                data_type_array.append(int(str_condition[num1]))
+        else:
+            return -1
+
```
Comment (Billy4195) [inline server_api.py]: extra blank line

File: server_api.py [hunk 4]
```diff
@@ -18,12 +19,161 @@
...
+        #insert infromation to schedule
+        send_msg["next_sn"] = 1
+        send_msg["target_id"] = [target_id]
+        send_msg["display_time"] = [display_time]
+        send_msg["arrange_sn"] = 0
+        receive_msg = edit_schedule(send_msg)
+        if receive_msg["result"]=="fail":
+            return -1
+
```
Comment (Billy4195) [inline server_api.py]: extra blank line

File: setting/server_setting.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+bluetooth_available 1
```
Comment (Billy4195) [inline setting/server_setting.txt]: Refer to #42, this file put in bluetooth will better than in setting.
And the "bluetooth_enable" is better than "bluetooth_available"

File: create_tables.sql [hunk 6]
```diff
@@ -98,3 +100,19 @@ create table arrange_mode
...
     armd_is_delete bit(1) default 0
 );
+
+
+DROP TABLE IF EXISTS user_prefer;
+
+create table user_prefer
+(
+        pref_id varchar(14) unique key,
+        user_id int default 0,
```
Comment (Billy4195) [inline create_tables.sql]: user_id should add NOT NULL attribute

Comment (icecat2012) [inline create_tables.sql]: we need to review all table 
I think this problem should be deal next time but not this merge

File: arrange_schedule.py [hunk 7]
```diff
@@ -1493,9 +1492,9 @@ def main():
                     os._exit(0)
                 else: #Parent
                     alarm_crawler_google_drive_img = raw_time + 3600.0
-            except:
+            except Exception as e:
                 receive_obj["result"] = "fail"
-                receive_obj["error"] = "fork6 error"
+                receive_obj["error"] = "fork6 error" + " : " + str(e)
                 set_system_log(receive_obj)
                 alarm_crawler_google_drive_img = raw_time + 600.0
```
Comment (Billy4195) [inline arrange_schedule.py]: One commit should do just one thing, if you want to add error message, please create another PR.

File: create_tables.sql (hunk 6 above)
Comment (Billy4195) [inline create_tables.sql]: Ok, this issue can be postponed

**Comment by Billy4195** [comment]
Please use `git rebase` to reduce the merge commit, and `git squash` to combine multiple commits to one commit.

**Comment by Billy4195** [comment]
Please solve the conflict, then push again

**Pull request merged**